lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -74,6 +74,35 @@ SmallVector<Direction> direction::unpackAttribute(Operation *component) {
 // Utilities
 //===----------------------------------------------------------------------===//
 
+/// Checks whether @p port is driven from within @p groupOp.
+static LogicalResult isPortDrivenByGroup(Value port, GroupOp groupOp,
+                                         bool checkInstanceOp = true) {
+  // Check if the port is driven by an assignOp from within @p groupOp
+  for (auto &use : port.getUses()) {
+    if (auto assignOp = dyn_cast<AssignOp>(use.getOwner()))
+      if (assignOp->getParentOfType<GroupOp>() == groupOp)
+        return success();
+  }
+
+  if (checkInstanceOp) {
+    // If the port is the result of an InstanceOp, and if any input port of
+    // this InstanceOp is driven within @p groupOp, we'll assume that @p port
+    // is sensitive to the driven input port.
+    if (auto instanceOp = dyn_cast<InstanceOp>(port.getDefiningOp())) {
```

Comment:
Maybe I'm missing something obvious, but I would either use `checkInstanceOp` and `.cast<InstanceOp>` or `dyn_cast` to see if it is actually an InstanceOp. I think the latter option is more succinct.

Nit: If possible, let's avoid nested control statement pyramids with `continue`.
E.g.

```
      for (auto portInfo : enumerate(compOpPortInfo)) {
        if (portInfo.value().direction == Direction::Output)
          continue;
        ...
      }
    }
```

Reply:
The function was also missing a check for whether a port was actually an output port of an instance. Given this, I've refactored it a bit, which also allowed me to remove the `checkInstanceOp` argument.